From d997978fc0c644784ed929b0c48e153d5c1f1408 Mon Sep 17 00:00:00 2001
From: Alexey Shvetsov <alexxyum@gmail.com>
Date: Sat, 19 Oct 2024 21:22:07 +0300
Subject: [PATCH] Fix build with torch if non default C{XX}_FLAGS set
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This should fix build colvars with torch if torch was built with non
default C{,XX}_FLAGS

Erorr info:

FAILED: src/gromacs/CMakeFiles/colvars_objlib.dir/__/external/colvars/colvar.cpp.o
/usr/bin/x86_64-pc-linux-gnu-g++ -DCOLVARS_TORCH -DGMX_DOUBLE=0 -I/usr/include/torch/csrc/api/include  -O2 -pipe -march=skylake -mtune=skylake -std=c++17 -fPIC "-O2 -pipe -march=skylake -mtune=skylake" -D_GLIBCXX_USE_CXX11_ABI=1 -fopenmp -MD -MT src/gromacs/CMakeFiles/colvars_objlib.dir/__/external/colvars/colvar.cpp.o -MF src/gromacs/CMakeFiles/colvars_objlib.dir/__/external/colvars/colvar.cpp.o.d -o src/gromacs/CMakeFiles/colvars_objlib.dir/__/external/colvars/colvar.cpp.o -c /var/tmp/portage/sci-chemistry/gromacs-9999/work/gromacs-9999/src/external/colvars/colvar.cpp
cc1plus: error: argument to ‘-O’ should be a non-negative integer, ‘g’, ‘s’, ‘z’ or ‘fast’

This caused by doubling quoted C{,XX}_FLAGS

Signed-off-by: Alexey Shvetsov <alexxyum@gmail.com>
---
 cmake/gmxManageColvars.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/gmxManageColvars.cmake b/cmake/gmxManageColvars.cmake
index 509d56e116..f49c820fa0 100644
--- a/cmake/gmxManageColvars.cmake
+++ b/cmake/gmxManageColvars.cmake
@@ -52,7 +52,7 @@ function(gmx_manage_colvars)
             target_link_libraries(colvars_objlib PRIVATE OpenMP::OpenMP_CXX)
         endif()
         if(GMX_TORCH)
-            target_compile_options(colvars_objlib PRIVATE ${CMAKE_CXX_FLAGS} ${TORCH_CXX_FLAGS})
+            target_compile_options(colvars_objlib PRIVATE ${CMAKE_CXX_FLAGS})
             target_include_directories(colvars_objlib PRIVATE ${TORCH_INCLUDE_DIRS})
             target_compile_definitions(colvars_objlib PRIVATE -DCOLVARS_TORCH)
         endif()
-- 
2.47.0

